task/blue-green-deploy/0.1/blue-green-deploy.yaml

```diff
@@ -0,0 +1,75 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: blue-green-deploy
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: deployment
+    tekton.dev/displayName: "blue green deployment"
+spec:
+  description: >-
+    This task can be used to do Blue-Green deployment
+  workspaces:
+    - name: manifest-dir
+      description: Consisting of kubernetes manifests
+    - name: kubeconfig-dir
+      description: Can be used in case kubeconfig file is mounted
+  params:
+    - name: SERVICE_NAME
+      type: string
+      description: Name of the service which is pointing to existing deployment
+    - name: NEW_VERSION
+      type: string
+      description: The version of newer deployment which is to be patched to existing service
+    - name: MANIFEST
+      type: string
+      default: "."
+      description: The newer version of the deployment to be deployed in another zone
+  steps:
+    - name: change-deployment
+      image: quay.io/openshift/origin-cli:latest
+      workingDir: $(workspaces.manifest-dir.path)
+      script: |
+        #!/usr/bin/env bash
+
```

Comment:
could you please add `set -eu -o pipefail`. It gives great confidences when tests are run against scripts that have -eu enabled.

Reply:
Hi @sthaha, I tried this but it's actually failing the whole script whenever it receives an error code `1` and I don't want it to fail atleast till the time `ProgressDeadlineExceeded` doesn't occurs

Reply:
-e is supposed to terminate whenever there is an error and you can ignore errors using ` || true` . Would that work?